### PR TITLE
Overview  UI performance

### PR DESCRIPTION
**Pull request by hanars**
Data for the project page is loaded in several separate requests. This was badly impacting the preformanceof the project page, as when each request completed any component that had the current project as a prop re-rendered. There were 3 categories of changes I made to address this:
1) Change the response from the server to remove the `<families/overview/etc>Loaded` field from the project response, and instead keep track of which entities have been loaded directly in the state, removing unnecessary updates to the project object
2) Instead of connecting several props to a top level component to pass down to its children, connecting the children directly and removing props from the parent, so when a part of the data updates only the relevant child component updates (i.e. instead of connecting the tag type counts to the top level project page, connect them directly to the tag summary component)
3) Instead of using the full project as a prop, only pass through the fields actually used by the relevant component.

### CODE REVIEW

**Review by ShifaSZ — CHANGES_REQUESTED** [review]

File: ui/pages/Project/components/ProjectCollaborators.jsx [hunk 1]
```diff
@@ -162,39 +162,40 @@ const collaboratorDisplay = ({ displayName, email }) => (
 
 const groupNameDisplay = ({ name }) => name
 
-const ProjectCollaborators = React.memo((
-  { project, user, loading, load, onSubmit, onGroupSubmit, addCollaborator },
-) => {
-  const canEdit = project.canEdit && !user.isAnvil
+const ProjectCollaborators = React.memo(({
+  canEdit, workspaceName, collaborators, collaboratorGroups, user, loading, load, onSubmit, onGroupSubmit,
+  addCollaborator,
```
[the remark this answers is not in the document]
Comment (hanars) [reply ui/pages/Project/components/ProjectCollaborators.jsx:167]: I added log statements for when this component was rendering and before this change it re-rendered when the project's locus lists were updated and after this change it didn't. So I don't know if react.memo is being more clever with prop comparison then just reference, but I am 100% confidant that this change improves performance

File: ui/pages/Project/components/ProjectOverview.jsx [hunk 2]
```diff
@@ -189,28 +191,31 @@ const FamiliesIndividualsOverview = connect(mapFamiliesStateToProps)(FamiliesInd
 
 const DataLoadedFamiliesIndividualsOverview = connect(mapDataLoadedFamiliesStateToProps)(FamiliesIndividuals)
 
-const MatchmakerOverview = React.memo(({ project }) => (
+const MatchmakerOverview = React.memo(({ name, mmeSubmissionCount, mmeDeletedSubmissionCount, canEdit }) => (
```
[the remark this answers is not in the document]
Comment (hanars) [reply ui/pages/Project/components/ProjectOverview.jsx:194]: per above, I explicitly tested the performance updates I was making, I didn't just like get an idea and implement it without bothering to see if it did anything. Every single change in this PR was confirmed to have a performance impact

File: ui/pages/Project/components/ProjectOverview.jsx [hunk 3]
```diff
@@ -373,41 +386,95 @@ LoadingSection.propTypes = {
   children: PropTypes.node,
 }
 
-const ProjectOverview = React.memo(({ familiesLoading, overviewLoading, ...props }) => (
+const ProjectOverview = React.memo(({
+  familiesLoading, overviewLoading, analysisGroupGuid, name, genomeVersion, workspaceName, workspaceNamespace,
+  canEdit, hasCaseReview, isAnalystProject, mmeSubmissionCount, mmeDeletedSubmissionCount,
```
[the remark this answers is not in the document]
Comment (hanars) [reply ui/pages/Project/components/ProjectOverview.jsx:391]: no. 6 renders on page load for one component is not worth it, even if the rendering component is not that big

File: ui/pages/Project/components/ProjectCollaborators.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShifaSZ) [reply ui/pages/Project/components/ProjectCollaborators.jsx:167]: It makes sense. Passing the whole `project` needs more re-rendering and impacts the performance. I thought it was the data-maintaining costing the performance.

File: ui/pages/Project/components/ProjectOverview.jsx [hunk 4]
```diff
@@ -373,41 +386,95 @@ LoadingSection.propTypes = {
...
+    workspaceName,
+    workspaceNamespace,
+    canEdit,
+    hasCaseReview,
+    isAnalystProject,
+    mmeSubmissionCount,
+    mmeDeletedSubmissionCount,
+  } = getCurrentProject(state)
+  return {
+    name,
```
Comment (ShifaSZ) [inline ui/pages/Project/components/ProjectOverview.jsx:468]: You may want to change the `name` to `projectName`.

**Review by ShifaSZ — APPROVED** [review]
Great changes.

**Pull request merged**